protocol/contracts/nexus/Shard.sol

```diff
@@ -0,0 +1,489 @@
+    }
+
+    /*
+     * @notice Delete recipe
+     * @param recipeId The recipe ID
+     */
+    function deleteRecipe(uint256 recipeId) external onlyElevatedAccess {
+        if (recipes[recipeId].inputShardIds.length == 0) { revert CommonEventsAndErrors.InvalidParam(); }
+        delete recipes[recipeId];
+        emit RecipeDeleted(recipeId);
+    }
+
+    /*
+     * @notice Set uri string of shard ID
+     * @param shardId The shard ID
+     * @param uri The uri string
+     * @return String uri of the shard ID
+     */
+    function setShardUri(uint256 shardId, string memory _uri) external onlyElevatedAccess {
+        if (bytes(_uri).length == 0 ) { revert CommonEventsAndErrors.InvalidParam(); }
+        shardUris[shardId] = _uri;
+        emit ShardUriSet(shardId, _uri);
+    }
+
+    /*
+     * @notice Get uri string of shard ID
+     * @param shardId The shard ID
+     * @return String uri of the shard ID
+     */
+    function uri(uint256 shardId) public view override returns (string memory) {
+        return shardUris[shardId];
+    }
+
+    /*
+     * @notice Transmute caller shards to create a new shard using a recipe
+     * Caller shards are burned and new shard(s) are minted to caller.
+     * @param recipeId The ID of the recipe
+     */
+    function transmute(uint256 recipeId) external {
+        address caller = msg.sender;
+        Recipe memory recipe = recipes[recipeId];
+        if (recipe.inputShardIds.length == 0) { revert InvalidRecipe(recipeId); }
+        /// @dev function checks caller has enough balances
+        _burnBatch(caller, recipe.inputShardIds, recipe.inputShardAmounts);
+        _mintBatch(caller, recipe.outputShardIds, recipe.outputShardAmounts, "");
+
+        emit Transmuted(caller, recipeId);
+    }
+
+    /*
+     * @notice Mint shard. This is a guarded function which only allows callers with perms to mint.
+     * Function checks if receiving mint address is blacklisted by Relic contract.
+     * If caps is set on shard ID for caller, new balance is checked.
+     * @param to The address to mint to
+     * @param shardId The shard ID
+     * @param amount The amount of shard ID to mint
+     */
+    function mint(
+        address to,
+        uint256 shardId,
+        uint256 amount
+    ) external isNotBlacklisted(to) {
+        if (!allowedShardIds[msg.sender].contains(shardId)) { revert CannotMint(shardId); }
+        /// @dev fail early
+        if (amount == 0) { revert CommonEventsAndErrors.InvalidParam(); }
+        // default cap 0 value means uncapped
+        uint256 cap = allowedShardCaps[msg.sender][shardId];
+        uint256 newBalance = mintBalances[msg.sender][shardId] + amount;
+        if (cap > 0 && newBalance > cap) {
+            revert MintCapExceeded(cap, newBalance);
+        }
+        mintBalances[msg.sender][shardId] = newBalance;
+        totalShardMints[shardId] += amount;
+        _mint(to, shardId, amount, "");
+        /// @dev track event with TransferSingle
```

Comment:
old comment? Want a separate event like you do with Relic?

Reply:
I removed the Relic event like you said. so will keep this as it is for consistency. So not an old comment